ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/routing/RoutingHooks.kt

```diff
@@ -0,0 +1,22 @@
+package io.ktor.server.routing
+
+import io.ktor.server.application.*
+
+/**
+ * Hook that will be triggered when a call was routed and before any other handlers.
+ */
+public object CallRouted : Hook<(RoutingApplicationCall) -> Unit> {
+    override fun install(application: ApplicationCallPipeline, handler: (RoutingApplicationCall) -> Unit) {
+        application.environment?.monitor?.subscribe(Routing.RoutingCallStarted) { call -> handler(call) }
+    }
+}
+
+/**
+ * Hook that will be triggered after a call processing was completely finished (i.e. all handlers were
+ * executed and response was sent)
```

Comment:
It's not guaranteed the response was sent here. It's possible that the call was not handled by routing and the engine will reply with 404 later in the pipeline

Reply:
Nice point, thanks!
Not sure about the "later in the pipeline" thing, though. As from what I can see, `RoutingCallFinished` event will be triggered after the whole pipeline was execcuted, in the `finally` block.

Reply:
Correct me if I'm wrong

Reply:
I would say:
```
Hook that will be triggered after a call processing was completely finished (i.e. all handlers were executed and response was sent or an error happened)
 ```

Reply:
It can be confusing, but we have 2 pipelines here: application and routing. Routing pipeline is executed inside `Call` phase of the application pipeline. So, this `finally` block is only at the end of the routing pipeline, but is still inside `Call` phase of the application pipeline. 
So, the correct doc will say that this hook is executed after routing was processed or error in handler happened.